### PR TITLE
[WIP] Use Accept header for Content-Type of response

### PR DESCRIPTION
**Pull request by winniehell**
Current behavior is that server decides what MIME types [are acceptable][mime-types] and which of those are preferred. Shouldn't this really be the business of the client and the `Accept`-header?

If Swagger definition and `Accept`-header of client agree on `image/jpeg`, but an object should be sent,  `application/json` is used anyway. In such a situation, I would rather prefer `406 Not Acceptable`.

This influences how I would proceed on #23.

[mime-types]: https://github.com/BigstickCarpet/swagger-express-middleware/blob/eb417f90d211cbcf9864345cfebbb9c1ec84fd45/lib/mock/index.js#L235

### CODE REVIEW

**Comment by winniehell** [comment]
In other words: what is the reason to have hard-coded MIME types in the code rather than relying on `produces` and `Accept`?

**Comment by winniehell** [comment]
@BigstickCarpet: Do you already have any opinion on that? I think that this is a prerequisite for #23. So I'd rather not continue work on that.

**Comment by BigstickCarpet** [comment]
Yeah, you're right.  The [`setContentType()`](https://github.com/BigstickCarpet/swagger-express-middleware/blob/eb417f90d211cbcf9864345cfebbb9c1ec84fd45/lib/mock/index.js#L315) method will need some intelligent logic where it chooses the best content type based on the `supported` and `excluded` params _as well as_ [`req.accepts()`](http://expressjs.com/en/api.html#req.accepts)

**Comment by winniehell** [comment]
Thank you for the feedback! I'll try to come up with some intelligent logic then... :smile:

**Comment by winniehell** [comment]
Currently, I don't agree that the `excluded` parameter in `setContentType()` is correct. I'll open a new pull request to keep this one clean.

File: lib/mock/index.js [hunk 1]
```diff
@@ -139,6 +152,10 @@ function mockResponseHeaders(req, res, next) {
 
     if (res.swagger.headers) {
       _.forEach(res.swagger.headers, function(header, name) {
```
Comment (BigstickCarpet) [inline lib/mock/index.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (14)](https://www.codacy.com/app/jamesmessinger/swagger-express-middleware/file/2175183918/issues/source?bid=2170795&fileBranchId=3076516#l154)

File: lib/mock/index.js [hunk 2]
```diff
@@ -137,50 +150,57 @@ function mockResponseHeaders(req, res, next) {
...
-              }
-              else {
-                res.set(name, _(sample).toString());
-              }
+    if (_.isObject(res.swagger.headers)) {
+      _(res.swagger.headers)
+        .omit(function (header, name) {
+          return !_.isObject(header) || 
+            !_.includes(['string', 'number', 'integer', 'boolean', 'array'], header.type) || 
+            !_.isString(name)
```
Comment (BigstickCarpet) [inline lib/mock/index.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/jamesmessinger/swagger-express-middleware/file/2175673392/issues/source?bid=2170795&fileBranchId=3076516#l158)

Comment (winniehell) [inline lib/mock/index.js]: Fixed. However I don't agree with this rule. :wink: See for example https://github.com/feross/standard

**Comment by winniehell** [comment]
I won't finish this as we have decided not to use swagger.

**Pull request closed without merging**